src/main/java/uk/ac/ebi/ena/readtools/loader/fastq/DataSpot.java

```diff
@@ -238,26 +238,26 @@
         {
             case CASAVA18:
                 if( !params.m_casava_1_8_name.reset( line ).find() )
-                    throw new DataFeederException( params.line_no, String.format( "Line [%s] does not match %s regexp", line, ReadStyle.CASAVA18 ) );
+                    throw new DataProducerException( params.line_no, String.format( "Line [%s] does not match %s regexp", line, ReadStyle.CASAVA18 ) );
                 bname = String.format( "%s/%s", params.m_casava_1_8_name.group( 1 ), params.m_casava_1_8_name.group( 2 ) );
                 break;
             
             case FASTQ:
                 if( !params.m_base_name.reset( line ).find() )
-                    throw new DataFeederException( params.line_no, String.format( "Line [%s] does not match %s regexp", line, ReadStyle.FASTQ ) );
+                    throw new DataProducerException( params.line_no, String.format( "Line [%s] does not match %s regexp", line, ReadStyle.FASTQ ) );
                 
                 bname = params.m_base_name.group( 1 );
                 break;
                 
             default:
-                throw new DataFeederException( params.line_no, String.format( "Line [%s] has no read style defined!", line ) );
+                throw new DataProducerException( params.line_no, String.format( "Line [%s] has no read style defined!", line ) );
         }
     }
     
     
     // get bases
     public void 
-    feedBases( InputStream is ) throws IOException, DataFeederException
+    feedBases( InputStream is ) throws IOException, DataProducerException

```

Comment:
Remove feedBases. I don't think it is used for anything except in DataSpotTest:testInvalidBaseCharactersException.